lib/vtls/schannel.c

```diff
@@ -1181,6 +1181,10 @@ schannel_connect_step2(struct connectdata *conn, int sockindex)
         failf(data, "schannel: SNI or certificate check failed: %s",
               Curl_sspi_strerror(sspi_status, buffer, sizeof(buffer)));
         return CURLE_PEER_FAILED_VERIFICATION;
+      case SEC_E_UNTRUSTED_ROOT:
+        failf(data, "schannel: untrusted root: %s",
+              Curl_sspi_strerror(sspi_status, buffer, sizeof(buffer)));
+        return CURLE_SSL_PEER_CERTIFICATE;
```

Comment:
```suggestion
        return CURLE_PEER_FAILED_VERIFICATION;
```
legacy

Reply:
Ugh, yes, thanks!

Reply:
another thing to consider is sspi strerror is very detailed, so the message is now effectively this (english version):

> schannel: untrusted root: SEC_E_UNTRUSTED_ROOT (0x80090325) - The certificate chain was issued by an authority that is not trusted.

which is a little redundant but fine. In some cases we have instead pointed out what API function has failed (in this case InitializeSecurityContext) and then sspi strerror gets the error.

Reply:
I don't think showing the API function will help many libcurl users. How about we cut out the "untrusted root" thing I added and we just make it?
~~~c
        failf(data, "schannel: %s",
              Curl_sspi_strerror(sspi_status, buffer, sizeof(buffer)));

Reply:
works for me